### PR TITLE
refactor: Refactor the way we generate suppressions

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/Content/Uno.UI.SourceGenerators.props [hunk 1]
```diff
@@ -202,25 +202,11 @@
...
 		<XamlGeneratorAnalyzerSuppressions Include="csharp-618 // Ignore obsolete members warnings" />
 		<XamlGeneratorAnalyzerSuppressions Include="csharp-105 // Ignore duplicate namespaces" />
 		<XamlGeneratorAnalyzerSuppressions Include="csharp-1591 // Ignore missing XML comment warnings" />
 		<XamlGeneratorAnalyzerSuppressions Include="csharp-CS8669 // Ignore annotation for nullable reference types" />
-
-		<!--
-		Ignore disposable fields, Uno manages those on its own.
-		https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1001
-		-->
-		<XamlGeneratorAnalyzerSuppressions Include="dotnet-CA1001" />
```
Comment (jeromelaban) [inline src/SourceGenerators/Uno.UI.SourceGenerators/Content/Uno.UI.SourceGenerators.props:223]: I think we should keep this one?

**Review by jeromelaban — APPROVED** [review]

File: src/SourceGenerators/Uno.UI.SourceGenerators/Content/Uno.UI.SourceGenerators.props (hunk 1 above)
Comment (Youssef1313) [reply src/SourceGenerators/Uno.UI.SourceGenerators/Content/Uno.UI.SourceGenerators.props:223]: Nope. Not needed. The analyzer does not report diagnostics on generated code, and Roslyn respects the auto-generated comment that we added at some point.

https://github.com/dotnet/roslyn-analyzers/blob/81d9b5d58e3d384ed4a91b75f4a1ece6352e76d9/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/TypesThatOwnDisposableFieldsShouldBeDisposable.cs#L41

**Review by ajpinedam — APPROVED** [review]
Good! :)

**Comment by Youssef1313** [comment]
@Mergifyio backport release/stable/5.3

**Pull request merged**